### ISSUE TITLE
Saved ODocument cannot be retrieved using the Index before commit in remote storage. But, works find in plocal

### ISSUE THREAD

**Issue by adithyank** [orientechnologies/orientdb#3139]
The document saved cannot be retrieved using the Index reference before commit in remote storage. But, it is working in plocal mode.

In remote mode, the variable retrievedId is null..

I have filed a similar issue #3018 for OrientVertex. It may be due to same issue as the underlying database is same.
1. What is the solution currently available to work-around this situation. or
2. is it a bug? If yes, in which release it will be resolved?

``` java
    private static void write() throws Exception
    {
        ODatabaseDocumentTx db = db();
        db.begin();

        /**************node creation starts**********/
        String KEY = "name-1";
        Map<String, Object> props = new HashMap<>();
        props.put("date", "" + new Date());
        props.put("name", KEY);
        props.put("prop1", "prop1-" + KEY);
        props.put("prop2", "prop2-" + KEY);

        ODocument doc = new ODocument("docnode");
        doc.fromMap(props);
        doc.save();
        /**************node creation ends**********/


        /**************get starts**********/
        final OIndexManagerProxy im = db.getMetadata().getIndexManager();
        OIndex index = im.getIndex(DATATYPE + "." + KEYFIELD);

        ORecordId retrievedId = (ORecordId) index.get(KEY);
        ODocument retrievedRec = retrievedId.getRecord();
        System.out.println("Retrieved doc before commit = " + retrievedRec.toMap());

        /**************get ends**********/

        System.out.println("node added");
        db.commit();
    }
```

**Comment by laa** [comment]
Assigned to @tglman

**Comment by tglman** [comment]
Hi,

This is implemented in 3.0  and released in 3.0-RC1, closing

Regards
